### ISSUE TITLE
`clang-tidy` fails on CXX modules

### ISSUE THREAD

**Issue by LecrisUT** [llvm/llvm-project#62258]
I don't know where the issue is coming from, so I am posting it here too. There is an equivalent [discourse on cmake](https://discourse.cmake.org/t/clang-tidy-cxx-modules/7938).

Error:
```
error: unable to handle compilation, expected exactly one compiler job in '' [clang-diagnostic-error]
warning: /clang-tidy-issue/test_module.ixx: 'linker' input unused [clang-diagnostic-unused-command-line-argument]
Found compiler error(s).
```

MWE:
```cmake
# CMakeLists.txt
cmake_minimum_required(VERSION 3.26)
project(clang-tidy-issue
		LANGUAGES CXX)

set(CMAKE_CXX_STANDARD 20)
set(CMAKE_EXPERIMENTAL_CXX_MODULE_CMAKE_API 2182bf5c-ef0d-489a-91da-49dbc3090d2a)
set(CMAKE_EXPERIMENTAL_CXX_MODULE_DYNDEP TRUE)
set(CMAKE_CXX_EXTENSIONS OFF)
set(CMAKE_EXPERIMENTAL_CXX_MODULE_MAP_FORMAT clang)
set(CMAKE_CXX_CLANG_TIDY "clang-tidy;-checks=*")

add_library(test_library)
target_sources(test_library PUBLIC
		FILE_SET cxx_modules
		TYPE CXX_MODULES
		FILES
		test_module.ixx
		)
```
```cpp
// test_module.ixx
module;

#include <string>

export module test_module;

export namespace test_module {
	consteval std::string_view version(){
		return "0.0.0";
	}
}
```

**Comment by LecrisUT** [comment]
So the issue here was that `.ixx` format was not being picked up. I have tested the current `main` branch that has 5b37cddff8e0140420ad776066529cf8f41d64d2, but this issue persists.

**Comment by carlosgalvezp** [comment]
Hi! That global option is only used in a handful of checks, and it's only used to determine whether a diagnostic belongs to a source file or a header file. Your error happens way before any of that. Since you are getting a Clang error (`clang-diagnostic-error`) I believe it makes sense for the Clang folks to take a look.

To clarify - are you able to compile the code with Clang (i.e. skipping clang-tidy)?

**Comment by LecrisUT** [comment]
> To clarify - are you able to compile the code with Clang (i.e. skipping clang-tidy)?

Yes, the code compiles just fine using cmake+ninja+clang, independent of the extension used (i.e. `.ixx` works just fine), it's only clang-tidy that fails to detect anything outside `.cxx`, `.cppm`, etc. extensions.

**Comment by zurrutik** [comment]
Are there any new developments regarding this?

**Comment by ChuanqiXu9** [comment]
> Are there any new developments regarding this?

In practice, we run clang-tidy after build so that clang-tidy can find them.
